phd-tests/framework/src/guest_os/windows_server_2022.rs

```diff
@@ -9,14 +9,24 @@
 
 use super::{CommandSequence, CommandSequenceEntry, GuestOs};
 
+/// Provides a guest OS adapter for Windows Server 2022 images. This adapter
+/// assumes the following:
+///
+/// - The image has been generalized (by running `sysprep /generalize`) and is
+///   configured so that on first boot it will skip the out-of-box experience
+///   (OOBE) and initialize the local administrator account with the appropriate
+///   password. See [MSDN's Windows Setup
+///   documentation](https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/generalize?view=windows-11)
+///   for more details.
```

Comment:
this is lovely, thank you for expanding on it!

Reply:
My pleasure! One additional detail here (I mentioned this in DMs, but adding here as well for posterity) is that the (WIP) Windows image builder currently generates generalized images (so that we don't have to worry about any specialization conflicts if they're used as the read-only base images for multiple VMs), and I leveraged that here to create the test image I used to test this adapter. It might be possible to create an image that's technically been specialized already (so that Setup doesn't have to run and we don't have to wait for this reboot) but can still be reused across multiple VMs, but I'd have to do some research to figure out how.